src/components/ResetPassword.js

```diff
@@ -0,0 +1,54 @@
+// ForgotPassword.js
+import React, { useState } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { resetPassword } from '../apis/authApi'; // Make sure to import your API call function
+
+const ResetPassword = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const token = searchParams.get('token');
+
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [error, setError] = useState('');
+
+  const handleResetPassword = async () => {
+    try {
+      if (password !== confirmPassword) {
+        setError('Passwords do not match');
+        return;
+      }
+
+      await resetPassword({ token, password });
```

Comment:
The `resetPassword` function is expecting an object with properties named: `passwordToken` and `newPassword`, but here you're passing an object with the properties `token` and `password`. If you stick a console.log in `resetPassword` you'll see that it's not getting those value because of this mismatch.

You can also see in the dev tools in the Network tab that nothing is being sent to the backend:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/1f8cc0ed-bb16-4a47-8601-b68ee65447da)

Reply:
Remember that when we write an object with just one value, it's a shortcut for saying that the `key` and the `value` have the same name. So what you've written is actually doing: `{ token: token, password: password}`

You can either name the keys the way the function expects them here: `await resetPassword({ passwordToken: token, newPassword: password })` .

Or you can update the function to read the keys that you're passing it. In that case, you still need to make sure that the object you're passing as the request body has the keys that the server is expecting:

```javascript
export const resetPassword = async ({ token, password }) => {
  const response = await axios.post('/auth/resetPassword', {
    passwordToken: token,
    newPassword: password,
  });
  return response.data;
};
```